src/pytorch_lightning/loggers/tensorboard.py

```diff
@@ -36,6 +35,17 @@
 from pytorch_lightning.utilities.logger import _sanitize_params as _utils_sanitize_params
 from pytorch_lightning.utilities.rank_zero import rank_zero_only, rank_zero_warn
 
+_TENSORBOARD_AVAILABLE = RequirementCache("tensorboard>=2.9.1")
+_TENSORBOARDX_AVAILABLE = RequirementCache("tensorboardX>=2.0")
+if TYPE_CHECKING:
+    # assumes at least one will be installed when type checking
+    if _TENSORBOARD_AVAILABLE:
+        from torch.utils.tensorboard import SummaryWriter
+    else:
+        from tensorboardX import SummaryWriter  # type: ignore[no-redef]
+else:
+    SummaryWriter = Any
```

Comment:
how this can happen if TBX is a mandatory dependency?

Reply:
but why is TBX is required rependency

Reply:
Importing has an overhead. You don't want to import the library unless you are going to use it. We use the `TYPE_CHECKING` flag here because forward references need to have a reference to the class.